### ISSUE TITLE
Not support Thermos.

### ISSUE THREAD

**Issue by azbh111** [IntellectualSites/PlotSquared#1448]
THermos: https://tcpr.ca/downloads/thermos
GitHub: https://github.com/CyberdyneCC/Thermos
PlotSquared: PlotSquared-Bukkit   3.5.0
Server start:

[09:37:09] [Server thread/INFO]: [PlotSquared] Enabling PlotSquared v3.5.0
[09:37:11] [Server thread/ERROR]: [PlotSquared] Plugin PlotSquared v3.5.0 has failed to register events for class com.plotsquared.bukkit.listeners.EntitySpawnListener because org/bukkit/event/entity/EntitySpawnEvent does not exist.
[09:37:11] [Server thread/INFO]: Please enable metrics for PlotSquared. Using metrics improves plugin stability, performance, and features. Bug fixes and new features are influenced on metrics.
[09:37:11] [Server thread/INFO]: [P2]  PlotSquared is using online UUIDs
[09:37:12] [Server thread/INFO]: class com.plotsquared.bukkit.util.SendChunk does not support ( 1,7,10 )
[09:37:13] [Server thread/INFO]: [P2] Key: plot-expiry, Value: false
[09:37:13] [Server thread/INFO]: [P2] Key: kill-road-mobs, Value: false
[09:37:13] [Server thread/INFO]: [P2] Key: worldedit-restrictions, Value: true
[09:37:13] [Server thread/INFO]: [P2] Key: database-purger, Value: false
[09:37:13] [Server thread/INFO]: [P2] Key: economy, Value: false
[09:37:13] [Server thread/INFO]: [P2] Key: block-cache, Value: true
[09:37:13] [Server thread/INFO]: [P2] Key: kill-road-vehicles, Value: false
[09:37:13] [Server thread/INFO]: [P2] Key: persistent-meta, Value: true
[09:37:13] [Server thread/INFO]: [P2] Key: ban-deleter, Value: false
[09:37:13] [Server thread/INFO]: [P2] Key: updater, Value: true
[09:37:13] [Server thread/INFO]: [P2] Key: database, Value: true
[09:37:13] [Server thread/INFO]: [P2] Key: plotme-converter, Value: false
[09:37:13] [Server thread/INFO]: [P2] Key: permission-cache, Value: true
[09:37:13] [Server thread/INFO]: [P2] Key: rating-cache, Value: true
[09:37:13] [Server thread/INFO]: [P2] Key: comment-notifier, Value: false
[09:37:13] [Server thread/INFO]: [P2] Key: metrics, Value: false
[09:37:13] [Server thread/INFO]: [P2] Key: chunk-processor, Value: false
[09:37:13] [Server thread/INFO]: [P2] Key: uuid-cache, Value: true
[09:37:13] [Server thread/INFO]: [P2] Key: events, Value: true
[09:37:13] [Server thread/INFO]: [P2] Key: commands, Value: true
[09:37:13] [Server thread/INFO]: PlotSquared is now enabled

create a new plotworld:

[09:47:58] [Server thread/INFO]: A Folder/World already exists with this name!
[09:47:58] [Server thread/INFO]: If you are confident it is a world you can import with /mvimport
[09:47:58] [Server thread/INFO]: Loading dimension 2 (plotworld) (net.minecraft.server.dedicated.DedicatedServer@d458084)
[09:47:58] [Server thread/INFO]: [P2] Detected world load for 'plotworld'
[09:47:58] [Server thread/INFO]: [P2]  - generator: PlotSquared>PlotSquared
[09:47:58] [Server thread/INFO]: [P2]  - plotworld: com.intellectualcrafters.plot.generator.HybridPlotWorld
[09:47:58] [Server thread/INFO]: [P2]  - manager: com.intellectualcrafters.plot.generator.HybridPlotManager
[09:47:58] [Server thread/INFO]: [P2]  - schematic: false
[09:47:58] [Server thread/INFO]: World possibly already loaded: plotworld
[09:47:58] [Server thread/INFO]: Preparing start region for level 3 (Dimension: 2, Seed: -8105894686067566243)
[09:47:58] [Server thread/WARN]: java.lang.reflect.InvocationTargetException
[09:47:58] [Server thread/WARN]: 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[09:47:58] [Server thread/WARN]: 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
[09:47:58] [Server thread/WARN]: 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
[09:47:58] [Server thread/WARN]: 	at java.lang.reflect.Constructor.newInstance(Unknown Source)
[09:47:58] [Server thread/WARN]: 	at com.intellectualcrafters.plot.util.block.QueueProvider$1.getNewQueue(QueueProvider.java:15)
[09:47:58] [Server thread/WARN]: 	at com.intellectualcrafters.plot.util.block.GlobalBlockQueue.getNewQueue(GlobalBlockQueue.java:69)
[09:47:58] [Server thread/WARN]: 	at com.plotsquared.bukkit.generator.BukkitPlotGenerator$1.populate(BukkitPlotGenerator.java:59)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.world.gen.ChunkProviderServer.func_73153_a(ChunkProviderServer.java:460)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.world.chunk.Chunk.func_76624_a(Chunk.java:1201)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.world.gen.ChunkProviderServer.originalLoadChunk(ChunkProviderServer.java:311)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.world.gen.ChunkProviderServer.loadChunk(ChunkProviderServer.java:217)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.server.management.PlayerManager$PlayerInstance.<init>(PlayerManager.java:408)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.server.management.PlayerManager.func_72690_a(PlayerManager.java:121)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.server.management.PlayerManager.func_72683_a(PlayerManager.java:176)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.server.management.ServerConfigurationManager.func_72375_a(ServerConfigurationManager.java:304)
[09:47:58] [Server thread/WARN]: 	at thermos.thermite.ThermiteTeleportationHandler.transferPlayerToDimension(ThermiteTeleportationHandler.java:92)
[09:47:58] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_7_R4.entity.CraftPlayer.teleport(CraftPlayer.java:514)
[09:47:58] [Server thread/WARN]: 	at com.plotsquared.bukkit.object.BukkitPlayer.teleport(BukkitPlayer.java:119)
[09:47:58] [Server thread/WARN]: 	at com.intellectualcrafters.plot.commands.Setup.onCommand(Setup.java:278)
[09:47:58] [Server thread/WARN]: 	at com.intellectualcrafters.plot.commands.SubCommand.execute(SubCommand.java:27)
[09:47:58] [Server thread/WARN]: 	at com.plotsquared.general.commands.Command.execute(Command.java:337)
[09:47:58] [Server thread/WARN]: 	at com.intellectualcrafters.plot.commands.MainCommand.execute(MainCommand.java:259)
[09:47:58] [Server thread/WARN]: 	at com.intellectualcrafters.plot.commands.MainCommand.onCommand(MainCommand.java:134)
[09:47:58] [Server thread/WARN]: 	at com.plotsquared.bukkit.util.BukkitCommand.onCommand(BukkitCommand.java:30)
[09:47:58] [Server thread/WARN]: 	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44)
[09:47:58] [Server thread/WARN]: 	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:181)
[09:47:58] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_7_R4.CraftServer.dispatchCommand(CraftServer.java:740)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.network.NetHandlerPlayServer.func_147361_d(NetHandlerPlayServer.java:1370)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.network.NetHandlerPlayServer.func_147354_a(NetHandlerPlayServer.java:1107)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.network.play.client.C01PacketChatMessage.func_148833_a(C01PacketChatMessage.java:38)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.network.play.client.C01PacketChatMessage.func_148833_a(C01PacketChatMessage.java:53)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:245)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:181)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:1023)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:432)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:841)
[09:47:58] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:693)
[09:47:58] [Server thread/WARN]: 	at java.lang.Thread.run(Unknown Source)
[09:47:58] [Server thread/WARN]: Caused by: java.lang.ClassNotFoundException: net/minecraft/server/v1_7_R4/Block
[09:47:58] [Server thread/WARN]: 	at java.lang.Class.forName0(Native Method)
[09:47:58] [Server thread/WARN]: 	at java.lang.Class.forName(Unknown Source)
[09:47:58] [Server thread/WARN]: 	at com.intellectualcrafters.plot.util.ReflectionUtils.getRefClass(ReflectionUtils.java:215)
[09:47:58] [Server thread/WARN]: 	at com.plotsquared.bukkit.util.block.BukkitLocalQueue_1_7.<init>(BukkitLocalQueue_1_7.java:24)
[09:47:58] [Server thread/WARN]: 	... 38 more

**Comment by azbh111** [comment]
[10:05:14 INFO]: Code:
[10:05:14 INFO]: Class<?> clazz = net.minecraft.server.v1_7_R4.Block.class;
                          System.out.println("test:"+clazz.getName());
                          clazz = net.minecraft.server.v1_7_R4.World.class;
                          System.out.println("test:"+clazz.getName());
                          clazz = net.minecraft.server.v1_7_R4.Chunk.class;
                          System.out.println("test:"+clazz.getName());
                          clazz = org.bukkit.craftbukkit.v1_7_R4.CraftWorld.class;
                         System.out.println("test:"+clazz.getName());
[10:05:14 INFO]: Result:
[10:05:14 INFO]: test:net.minecraft.block.Block
[10:05:14 INFO]: test:net.minecraft.world.World
[10:05:14 INFO]: test:net.minecraft.world.chunk.Chunk
[10:05:14 INFO]: test:org.bukkit.craftbukkit.v1_7_R4.CraftWorld

**Comment by azbh111** [comment]
In CraftBukkit API-1.7.10,it should be org/bukkit/event/entity/CreatureSpawnEvent instead of org/bukkit/event/entity/EntitySpawnEvent

**Comment by MattBDev** [comment]
azbh111, Thermos has many quirks to it. It is also a bit more difficult to maintain support for it considering it is using 1.7.10 which was released 2 years ago (a long time). The event you suggested isn't right though. The event EntitySpawnEvent is a Spigot event possibly introduced after 1.7.10. I'll look into a fix for this issue. 
Also, I saw you disabled Metrics. Metrics helps us know if we have users using Thermos.
